slides/aml-25-042518-time-series/index.html

```diff
@@ -0,0 +1,952 @@
+      }
+      .wide-left-column {
+          float: left;
+          width: 65%
+      }
+      .invisible {
+          visibility: hidden
+      }
+      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
+    </style>
+  </head>
+  <body>
+    <textarea id="source">
+
+class: center, middle
+
+### W4995 Applied Machine Learning
+
+# Time Series and Forecasting
+
+04/25/18
+
+Andreas C. Müller
+
+???
+
+---
+class: spacious
+
+#What's different ?
+
+- Not iid
+
+
+???
+
+---
+
+
+# Equally spaced & not equally spaced
+
+`$$ (x_{t_1}, y_{t_1}), ... (x_{t_m}, y_{t_m})  \quad t_1 < t_2 < ... < t_m $$`
+
+Equally spaced:
+
+$$ t_{i+1} - t_i = const$$ 
+- 1 second, 1 day, 1 year
+
+Unevenly spaced:
+
+$$ t_{i+1} - t_i  \quad varies $$ 
+- Event time series (earthquake, trade, ...)
+
+
+???
+
+---
+
+
+.left-column[
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; Evenly spaced
+![:scale 95%](images/evenly spaced data.png)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CO2 Measurements
+- Today only evenly spaced
+
+]
+.right-column[
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Unevenly spaced
+![:scale 100%](images/unevenly spaced data.png)
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Volcanic Eruption Deaths
+- For uneven: sometimes aggregates over fixed periods make sense
+]
+
+???
+
+---
+class: center, middle
+
+# Tasks 
+
+???
+
+---
+
+# 1d Forecasting
+
+.center[![:scale 80%](images/1d forecasting.png)]
+
+???
+
+---
+
+# Nd Forecasting
+
+.center[![:scale 80%](images/nd forecasting.png)]
+
+???
+
+---
+class: spacious
+
+# Feature-based forecasting &nbsp; (expanatory variables)
+
+.center[![:scale 80%](images/feature based forecasting.png)]
+
+???
+
+---
+
+# Time Series Classification
+
+.center[![:scale 70%](images/time series classification.png)]
+
+???
+
+---
+
+
+# Tasks
+
+- 1d time-series forecasting
+  - Number of customers (naive)
+- Multivariate time-series -- forecast all
+  - Stock market
+- Multivariate time-series -- forecast one
+  - Number of customers (with features)
+- Time-series classification (1d or Nd)
+  - Activity from smart phone sensors
+
+???
+
+---
+class: center, middle
+
+# Data reading / munging with Pandas
+
+???
+
+---
+
+# Parse Dates
+
+.smaller[
+```python
+url = "ftp://aftp.cmdl.noaa.gov/products/trends/co2/co2_weekly_mlo.txt"
+names = ["year", "month", "day", "year_decimal", "co2", "days", "1 yr ago", 
+          "10 yr ago", "since 1800"]
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, na_values=[-999.99])
+maunaloa.head()
+```
+]
+
+.center[![:scale 80%](images/parse dates.png)]
+
+???
+
+---
+
+# Parse Dates
+
+.smaller[
+```python
+maunaloa = pd.read_csv(url, skiprows=49, header=None, delim_whitespace=True,
+                       names=names, parse_dates=[[0, 1, 2]], na_values=[-999.99])
```

Comment:
if you remove a note from the slide like here, an you please add it to the slide notes?

Reply:
done

Reply:
Actually most of the missing parts are due to me referring to the .odp file and not the notes.pdf

Reply:
I was looking at the odp file. Opening it in power point must have swallowed it.